UnityProject/Assets/Scripts/Core/Lighting System/Light2D/LightSprite.cs

```diff
@@ -25,6 +25,8 @@ public class LightSprite : CustomSprite
         private Vector3 _oldLightOrigin;
         private LightShape _oldLightShape;
 
+        public int GivenID = 0;
```

Comment:
might be better to make a system to handle spawning in any sprite, using sprite handler  system,  that a lot of work so it's okay just to do it for this